File: test/init.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { mkdtempSync, readFileSync, existsSync } from "node:fs";
import { tmpdir } from "node:os";
import { join } from "node:path";
import yaml from "js-yaml";
import { initArea } from "../brain_tools/init.js";
import { CURRENT_SCHEMA_VERSION } from "../brain_tools/schema.js";

function tmp() {
  return mkdtempSync(join(tmpdir(), "init-"));
}

test("init creates manifest and index", () => {
  const area = join(tmp(), "work");
  initArea(area, { areaName: "work", sensitivity: "confidential" });
  const manifest = yaml.load(readFileSync(join(area, ".brain.yml"), "utf-8"));
  assert.equal(manifest.area, "work");
  assert.equal(manifest.sensitivity, "confidential");
  assert.equal(manifest.schema_version, CURRENT_SCHEMA_VERSION);
  assert.ok(existsSync(join(area, "MOC.md")));
});

test("init stores description, layout and keywords", () => {
  const area = join(tmp(), "education");
  initArea(area, {
    areaName: "education",
    sensitivity: "personal",
    description: "Обучение",
    layout: "<it|english>/<topic>",
    keywords: ["обучение", "it"],
  });
  const m = yaml.load(readFileSync(join(area, ".brain.yml"), "utf-8"));
  assert.equal(m.description, "Обучение");
  assert.equal(m.layout, "<it|english>/<topic>");
  assert.deepEqual(m.keywords, ["обучение", "it"]);
});

test("init rejects invalid sensitivity", () => {
  const area = join(tmp(), "work");
  assert.throws(
    () => initArea(area, { areaName: "work", sensitivity: "secret" }),
    /invalid sensitivity/
  );
});

File: src/test/java/brain/InitTest.java
package brain;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.Map;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;

class InitTest {
  @Test
  void initCreatesManifestAndIndex(@TempDir Path root) throws Exception {
    Path area = root.resolve("work");
    Init.initArea(area, "work", "confidential", "", "<topic>", List.of());
    Map<String, Object> manifest = Yamls.load(Files.readString(area.resolve(".brain.yml")));
    assertEquals("work", manifest.get("area"));
    assertEquals("confidential", manifest.get("sensitivity"));
    assertEquals(Schema.CURRENT_SCHEMA_VERSION, manifest.get("schema_version"));
    assertTrue(Files.exists(area.resolve("MOC.md")));
  }

  @Test
  void initStoresDescriptionLayoutKeywords(@TempDir Path root) throws Exception {
    Path area = root.resolve("education");
    Init.initArea(area, "education", "personal", "Обучение",
        "<it|english>/<topic>", List.of("обучение", "it"));
    Map<String, Object> m = Yamls.load(Files.readString(area.resolve(".brain.yml")));
    assertEquals("Обучение", m.get("description"));
    assertEquals("<it|english>/<topic>", m.get("layout"));
    assertEquals(List.of("обучение", "it"), m.get("keywords"));
  }

  @Test
  void initRejectsInvalidSensitivity(@TempDir Path root) {
    Path area = root.resolve("work");
    assertThrows(IllegalArgumentException.class,
        () -> Init.initArea(area, "work", "secret", "", "<topic>", List.of()));
  }
}
